gsa/src/web/entities/bulktagcomponent.js

```diff
@@ -0,0 +1,216 @@
+/* Copyright (C) 2020 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useReducer, useEffect, useCallback} from 'react';
+
+import _ from 'gmp/locale';
+
+import {getEntityType, apiType, typeName} from 'gmp/utils/entitytype';
+
+import TagsDialog from 'web/entities/tagsdialog';
+
+import {useBulkTag, ENTITY_TYPES} from 'web/graphql/tags';
+
+import TagDialog from 'web/pages/tags/dialog';
+
+import SelectionType, {getEntityIds} from 'web/utils/selectiontype';
+import useGmp from 'web/utils/useGmp';
+import useUserSessionTimeout from 'web/utils/useUserSessionTimeout';
+
+const initialState = {
+  tag: {},
+  tags: [],
+  tagDialogVisible: false,
+};
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'setState':
+      const {newState} = action;
+      return {
+        ...state,
+        ...newState,
+      };
+    default:
+      return state;
+  }
+};
+
+const BulkTagComponent = ({
+  entities,
+  selected,
+  filter,
+  selectionType,
+  entitiesCounts,
+  onClose,
+}) => {
+  const gmp = useGmp();
+  const [, renewSession] = useUserSessionTimeout();
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const [bulkTag] = useBulkTag();
+
+  const getTagsByType = useCallback(
+    entities => {
+      if (entities.length > 0) {
+        const filter = 'resource_type=' + apiType(getEntityType(entities[0]));
```

Comment:
I would extract the entity type into a variable of the component because it is used several times.